frontend/src/views/SecretMainPage/components/CreateSecretForm/CreateSecretForm.tsx

```diff
@@ -50,12 +51,39 @@ export const CreateSecretForm = ({
   const { closePopUp } = usePopUpAction();
 
   const { mutateAsync: createSecretV3 } = useCreateSecretV3();
+  const createWsTag = useCreateWsTag();
+
   const { permission } = useProjectPermission();
   const canReadTags = permission.can(ProjectPermissionActions.Read, ProjectPermissionSub.Tags);
   const { data: projectTags, isLoading: isTagsLoading } = useGetWsTags(
     canReadTags ? workspaceId : ""
   );
 
+  const createNewTag = async (slug: string) => {
+    // TODO: Replace with slugSchema generic
+    const slugSchema = z.string().trim().toLowerCase().min(1);
+    try {
+      const parsedSlug = slugSchema.parse(slug);
+      await createWsTag.mutateAsync({
+        workspaceID: workspaceId,
+        tagSlug: parsedSlug,
+        tagColor: ""
+      });
+    } catch (error) {
+      if (error instanceof z.ZodError) {
+        createNotification({
```

Comment:
Mutation errors are automatically handled in `/src/reactQuery.tsx`, so this would result in multiple error notifications if the error is API related.

I would remove the `if (error instanceof z.ZodError) {` handler entirely